Datadog/IntegrationUnitTests/RUM/RUMViewScopeIntegrationTests.swift

```diff
@@ -0,0 +1,193 @@
+
+        let customView = try XCTUnwrap(session.views.first(where: { $0.name == viewName }))
+
+        customView.viewEvents.forEach { viewEvent in
+
+            XCTAssertEqual(viewEvent.context?.contextInfo as? [AttributeKey: AnyCodable], initialAttributes)
+        }
+    }
+
+    func testWhenViewRelyingOnGlobalAttributes_doesNotUpdateTheViewAttributes_afterStopView() throws {
+        // Given
+        let initialAttributes: [AttributeKey: AnyCodable] = ["key1": .init("value1"), "key2": .init("value2")]
+        let viewName = "MyView"
+        RUM.enable(with: rumConfig, in: core)
+
+        let monitor = RUMMonitor.shared(in: core)
+
+        // When
+        monitor.startView(key: "key", name: viewName)
+        monitor.addAttribute(forKey: "key1", value: initialAttributes["key1"])
+        monitor.addAttribute(forKey: "key2", value: initialAttributes["key2"])
+        monitor.stopView(key: "key")
+        // Flushing the commands in the current thread to prevent data races accessing the attributes
+        core.flush()
+        monitor.addAttribute(forKey: "additionalKey", value: "additionalValue")
+
+        // Then
+        let session = try RUMSessionMatcher
+            .groupMatchersBySessions(try core.waitAndReturnRUMEventMatchers())
+            .takeSingle()
+
+        // It should have the `ApplicationLaunchView` and `MyView` views
+        XCTAssertEqual(session.views.count, 2)
+
+        let customView = try XCTUnwrap(session.views.first(where: { $0.name == viewName }))
+
+        customView.viewEvents.forEach { viewEvent in
+
+            XCTAssertEqual(viewEvent.context?.contextInfo as? [AttributeKey: AnyCodable], initialAttributes)
+        }
+    }
+
+    func testWhenSessionEnds_theViewAttributesAreNotUpdated_withAStopView() throws {
+        // Given
+        let initialAttributes: [AttributeKey: AnyCodable] = ["key1": .init("value1"), "key2": .init("value2")]
+        let viewName = "MyView"
+        RUM.enable(with: rumConfig, in: core)
+
+        let monitor = RUMMonitor.shared(in: core)
+
+        // When
+        monitor.startView(key: "key", name: viewName, attributes: initialAttributes)
+        monitor.stopSession()
+        monitor.stopView(key: "key", attributes: ["additionalKey" : "additionalValue"])
+
+        // Then
+        let session = try RUMSessionMatcher
+            .groupMatchersBySessions(try core.waitAndReturnRUMEventMatchers())
+            .takeSingle()
+
+        // It should have the `ApplicationLaunchView` and `MyView` views
+        XCTAssertEqual(session.views.count, 2)
+
+        let customView = try XCTUnwrap(session.views.first(where: { $0.name == viewName }))
+
+        customView.viewEvents.forEach { viewEvent in
+
+            XCTAssertEqual(viewEvent.context?.contextInfo as? [AttributeKey: AnyCodable], initialAttributes)
+        }
+    }
+
+    func testWhenViewReceivesActions_theViewAttributesAreNotUpdated_withActionAttributes() throws {
```

Comment:
change-request/ This test seems unusual as it doesn't validate existing behavior but instead asserts that a non-existent behavior doesn't occur. Event attributes are not meant to be inherited by other events, which is intentional. This test essentially verifies that this behavior doesn't happen.

A more valuable approach would be to add coverage for global attributes, which should be included in all events. Specifically, it would be useful to test that:

- Global attributes are added to action events.
- Global attributes are added to resource events.
- Global attributes are added to error events.

Reply:
Yes, it is not a behaviour in the product but the idea of the test is to give us confidence we don't have regressions on this regard. 
The attributes are shared through commands for all types (actions, resources, views,..) and so it is very easy to have wrong assumptions by not knowing the playbook.

Reply:
As suggested I added tests to validate the addition of global attributes to actions, resources and errors